lib/wallet/wallet.js

```diff
@@ -1820,6 +1825,73 @@ class Wallet extends EventEmitter {
     }
   }
 
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX with a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   * @returns {{ bid: MTX; reveal: MTX;}} {bid, reveal}
+   */
+
+  async createAuctionTxs(name, value, lockup, options) {
+    const unlock = await this.fundLock.lock();
+    try {
+      return await this._createAuctionTxs(name, value, lockup, options);
+    } finally {
+      unlock();
+    }
+  }
+
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX without a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   * @returns {{ bid: MTX; reveal: MTX;}} {bid, reveal}
+   */
+
+  async _createAuctionTxs(name, value, lockup, options) {
+    const bid = await this._createBid(name, value, lockup, options);
+
+    const bidOuputIndex = bid.outputs.findIndex(o => o.covenant.isBid());
+    const bidOutput = bid.outputs[bidOuputIndex];
+    const bidCoin = Coin.fromTX(bid, bidOuputIndex, -1);
+
+    // Prepare the data needed to make the reveal in advance
+    const nameHash = bidOutput.covenant.getHash(0);
+    let ns = await this.getNameState(nameHash);
+    if (!ns)
+      ns = await this.wdb.getNameStatus(nameHash);
```

Comment:
Yeah I think this is right. If the wallet doesn't have a local copy of the namestate that means nothing has been confirmed in the wallet for this name: It doesn't have any bids yet, etc. So it calls `getNameStatus()` which actually reaches out the full node instead and tries to get the name's status from the blockchain.

HOWEVER, I don't think you need this at all, because you already got this info when you made the bid! If you look below, all that `ns` is used for is `ns.height`... which is already in the BID covenant!

You already have this above:
`const nameHash = bidOutput.covenant.getHash(0);`
Pretty sure you can follow that with:
`const height = bidOutput.covenant.getU32(1);`

and then you don't need to get namestate anymore here at all.

Reply:
Thanks for the tip, changed it